### ISSUE TITLE
How to display images on search results?

### ISSUE THREAD

**Issue by leonardldev6** [tmsss/gatsby-plugin-flexsearch#7]
Each post on my gatsby site has a photo, and I would like to be able to show that on my search results. How do I go about doing that?

So far I've added this to `gatsby-config.js`, and what it returns me on the search results page through `page.authorphoto` is `'../../../static/images/photo.jpg'` and if that is added to my src on an `<img />` tag, the photo shows as broken.

How do I handle this scenario so that the photo actually shows up on the page?

          {
            name: 'authorphoto',
            indexed: false,
            resolver: 'frontmatter.authorphoto',
            attributes: {
              encode: 'balance',
              tokenize: 'strict',
              threshold: 6,
              depth: 3
            },
            store: true
          }

**Comment by tmsss** [comment]
I haven't tested for that situation, but that is a good use case. What is the image src absolute path for the image that appears in the rendered post? You can try to catch the src from search results and transform with regex to the absolute path. If you paste here the code, maybe I can help you.

**Comment by hrishikesh-k** [comment]
I needed the exact functionality, too bad, the discussion doesn't really take me till the end. Well, I've images in `frontmatter` too and from `GraphiQL`, I found out the node responsible for giving me the image `frontmatter.coverImage.childImageSharp.fluid.originalImg`.

So, now, my gatsby-config.js is like this:

```
{
	resolve: 'gatsby-plugin-flexsearch',
	options:
		{
			languages: ['en'],
			type: 'MarkdownRemark',
			fields: [
				{
					name: 'title',
					indexed: true,
					resolver: 'frontmatter.title',
					store: true
				},
				{
					name: 'icon',
					indexed: false,
					resolver: 'frontmatter.coverImage.childImageSharp.fluid.originalImg',
					store: false
				},
				{
					name: 'url',
					indexed: false,
					resolver: 'frontmatter.path',
					store: true
				}],
		}
},
```

My search component currently is exactly the same as the one from Readme.md of this repo. Is it possible to show images alongside the results in this setup?

**Comment by hrishikesh-k** [comment]
After a lot of trial and errors or various changes to the above config, I've got some info:

The `flexsearch_index.json` only stores the value of `frontmatter.coverImage`, but, that path only contains a relative file path of the image from the `.md` file. If I try any other query like `frontmatter.coverImage.childImageSharp.fluid.src` (this is another node that's giving me image path in build in `GraphiQL`), the `flexsearch_index.json` just stores the other two entries (`title` and `url`). I've no idea how I can get this thing to store (and later use) the image paths from the build. Any help would be appreciated.

This is my latest config for the icon part:

```
{
  name: 'icon',
  indexed: false,
  resolver: 'frontmatter.coverImage.childImageSharp.fluid.originalImg',
  store: true
},
```

this gives me the index as:

```
  "id": #, //some number
  "node": {
    "title": "Title", //title of the page
    "url": "/path" //path of the page
        }
```

However, when I set my config like:

{
  name: 'icon',
  indexed: false,
  resolver: 'frontmatter.coverImage',
  store: true
},

I get...

```
  "id": #, //some number
  "node": {
    "title": "Title", //title of the page
    "icon": "../relative-path", //relative path of the image exactly as I have mentioned in the .md file
    "url": "/path" //path of the page
        }
```

That's the closest I've managed to reach in order to include image paths in the store. But, as far as I know, those relative paths are useless.

**Comment by hrishikesh-k** [comment]
I found something useful in the Lunr plugin for Gatsby. The discussion is here: https://github.com/humanseelabs/gatsby-plugin-lunr/issues/8. Is it possible in FlexSearch. If yes, how can I do it?

**Comment by hrishikesh-k** [comment]
I created a stupid workaround myself. I copied my images (converted to a smaller size) in my `static` directory, added a new line to my `frontmatter` called `searchIcon` and now, this field has the path of the images in the `static` folder, thus, relative to the domain. So, I modified my config for icons like this:

{
  name: 'icon',
  indexed: false,
  resolver: 'frontmatter.searchIcon',
  store: true
},

and my `search.js` like this:

```
<div className = {style.itemSearch} key = {i}>
  <Link to = {page.url} className = {style.link}>
    <img alt = "icon" src = {page.icon}/>
    <p>{page.title}</p>
  </Link>
</div>
```

I've adjusted my CSS styles accordingly and it works. The only thing, I have to do the same for all my `.md` files that I want to display icons for, but, that's the only non-programmer move I could think of.

Thanks for this thread, it somehow kept me inspired to look for a solution thinking it might be possible.
